jcommune-view/jcommune-web-controller/src/main/java/org/jtalks/jcommune/web/controller/AdministrationController.java

```diff
@@ -298,4 +307,28 @@ private void checkForAdminPermissions() {
         long forumId = componentService.getComponentOfForum().getId();
         componentService.checkPermissionsForComponent(forumId);
     }
+
+    private JsonResponse saveOrUpdateGroup(@Valid @RequestBody GroupAdministrationDto groupDto, BindingResult result, Locale locale) throws org.jtalks.common.service.exceptions.NotFoundException {
+        checkForAdminPermissions();
+        if (result.hasFieldErrors() || result.hasGlobalErrors()) {
+            return new JsonResponse(JsonResponseStatus.FAIL, result.getAllErrors());
+        }
+        try {
+            groupService.saveGroup(groupDto);
```

Comment:
I think it was decided this way during the discussion, to pass DTO to service, instead of building Entity in the controller. IMHO it's better to do it this way, bcs the Entity is the knowledge of service and persistence layer. There can be multiple fields, that are unknown for the controller, and some object construction tricks. With the DTO you are able to pass only required things, without digging into details.